### PR TITLE
Replace Travis with github actions.

### PR DESCRIPTION
**Pull request by modmuss50**
I tried to use https://github.com/marketplace/actions/run-java-checkstyle to comment on PRs with the checkstyle results but it seemed to fail to build even when it passed with no useful log output.

Should this also build against java 8? I dont see a big need for it, but it might be useful.

Once this is merged I will move this to the other projects that have travis (just loader I think). As for replacing jenkins with actions it might be possible.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Build
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 14
```
Comment (liach) [inline .github/workflows/build.yml]: github action probably has matrix that supports multiple java versions, though i don't know how to do such

Comment (modmuss50) [reply .github/workflows/build.yml]: Yes, I does. I will add j8 as well.

**Comment by liach** [comment]
Hmm, doesn't the action run with the current action config on the pr branch? does it always use action config from default branch? If it only uses the config from default branch, would be a bit painful to debug action prs.

**Comment by modmuss50** [comment]
Updated to build on 8 11 and 14.

Should be good to merge if no one has any objections.

**Comment by modmuss50** [comment]
It only builds from the main branch, I believe (but could be wrong) that travis does this. Seems more secure this way as well so I dont see it being an issue.

**Review by liach — APPROVED** [review]
Personally I don't want some robot github account commenting checkstyle errors out.

Given the strict style we have, some new contributor may have a lot of style problems, causing spammy style review comments that makes reviewing for regular user harder.

Also since our `build` task already fails if any of the checkstyle tasks fails, it should be enough to guard against bad code style. Moreover, these individual style problems should be available on action's log (which is public afaik).

**Comment by modmuss50** [comment]
@liach I agree, I kinda wanted to just test to see how it worked tbh,

**Comment by haykam821** [comment]
Shouldn’t [checks](https://developer.github.com/v3/checks/) be used rather than comments?

**Comment by liach** [comment]
Hmm, good question. I personally have never even heard of the check system. Seems reviewdog used by the checkstyle action plugin does have a [github checks sink](https://github.com/reviewdog/reviewdog#reporter-github-checks--reportergithub-pr-check). 

Feel free to either merge this as-is or test out github checks integration with the action plugin.

**Comment by modmuss50** [comment]
Yes, if you can get it to work feel free to PR it once this is in, or let me know

**Pull request merged**